core/src/main/scala/kafka/tools/ConsumerPerformance.scala

```diff
@@ -266,11 +280,12 @@ object ConsumerPerformance {
           val messageAndMetadata = iter.next
           messagesRead += 1
           bytesRead += messageAndMetadata.message.length
+          val currentTimeMillis = System.currentTimeMillis
```

Comment:
I am not sure how much percentage this would affect the evaluation results. This is probably fine. But if it does skew the result too much, we can solve the problem by scheduling `printProgressMessage ` at regular interval using `ScheduledExecutorService . scheduleAtFixedRate(...)`. A `lock `or `synchronize` can be used support concurrent access to related variables if this is a concern.

I am not suggesting to do it because the improved accuracy may not worth the added code complexity. Just want to raise the concern of potentially accuracy skew. Committer can recommend whether to do it or not.